### PR TITLE
Fix Schema validator in examples.test.ts

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: examples/examples.test.ts [hunk 1]
```diff
@@ -33,7 +40,7 @@ function validateAgainstSchemas(vlspec) {
   assert(isVgValid);
 }
 
-function getExampleList(examples) {
+export function getExampleList(examples) {
```
Comment (domoritz) [inline examples/examples.test.ts]: what's this function? Can you delete it?

Comment (kanitw) [inline examples/examples.test.ts]: Good point.  I was a function that you should have removed when we changed it to read directly from directory.


Done.

Comment (domoritz) [inline examples/examples.test.ts]: Hehe, yeah.

File: examples/examples.test.ts [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-const ZSchema = require('z-schema');
+const zSchema = require('z-schema');
 const inspect = require('util').inspect;
 const dl = require('datalib');
 const fs = require('fs');
 const path = require('path');
 
-const validator = new ZSchema({
+zSchema.registerFormat('color', function (str) {
+  return false;
```
Comment (domoritz) [inline examples/examples.test.ts]: Hmm, don't you have to actually validate the string? I'm surprised that the tests pass if the validator always returns false.

Comment (kanitw) [inline examples/examples.test.ts]: I just copied what you did earlier.

Comment (domoritz) [inline examples/examples.test.ts]: The difference is that from where you copied it, we only care that the validator exists. Here the format validator should actually check whether `red` is a color. 

I guess we never specify a color in an example?

File: test/util.ts [hunk 3]
```diff
@@ -8,3 +9,14 @@ import {Spec} from '../src/schema/schema';
 export function parseModel(spec) {
   return new Model(spec as Spec);
 }
+
+export const zSchema = require('z-schema');
+
+zSchema.registerFormat('color', function (str) {
+  // valid colors are in list or hex color
+  return contains(['purple'], str) || /^#([0-9a-f]{3}){1,2}$/i.test(str);
```
Comment (kanitw) [inline test/util.ts]: Why only purple?

Comment (kanitw) [inline test/util.ts]: Better copy a list of valid svg & canvas color name from somewhere?

Comment (domoritz) [inline test/util.ts]: Cause that's the only color we actually use :-P Feel free to add it. I don't think we need to be exhaustive.

**Comment by domoritz** [comment]
Fixes https://github.com/vega/vega-lite/issues/1156

**Pull request merged**